landside/catkin_ws/src/gui/README.md

```diff
@@ -86,7 +85,125 @@ The table will show you what nodes are running.
 
 To stop a node that was run using the Launch Dialog, you may click on the item in the table and, on your keyboard, press Delete for Windows or Fn + Delete for Mac, which will stop the node.
 
-Data is published in percentage for CPU and GB for RAM in the System Usage box.
+#### Launch Argument Validation & Help
+
+The Launch Plugin also has the capability to validate arguments input by the user before the node is launched and provide descriptions of arguments. 
+
+To use this capability, add the `doc` attribute to `<arg>` tag in your launch file. Inside this `doc` attribute, put a JSON object.
+
+Example:
+```xml
+<arg name="count" default="" doc='{"type":"int","allowEmpty"=false,"help":"The number of messages to publish"}'>
+```
+**There must NOT be anything other than the JSON object in the `doc` attribute for the validation to work.**
```

Comment:
I know that ideally we would add attributes to the XML tags to communicate additional information beyond what roslaunch supports. However, adding custom attributes to the XML makes roslaunch give warnings. For example, I added the `type` attribute to `framerate` argument in `usb_camera.launch` like so
```
<arg name="framerate" default="-1" type="int" />
```
and when launching the file the following warning printed out to the console:
```
WARNING: [/root/dev/robosub-ros/onboard/catkin_ws/src/cv/launch/usb_camera.launch] unknown <arg> attribute 'type'
```
While these are warnings and not fatal errors that stop the launch, I thought it would be quite a nuisance to have a bunch of warnings show up every time we launch a file with custom attributes in the XML. The only way  to add extra information to the `arg` tags without roslaunch printing warnings would be to add it to an attribute. The `doc` attribute is most appropriate for this as ROS does not try to interpret the `doc` attribute and so we are free to put any information there.

An additional benefit of adding information to the `doc` attribute is that it is printed out with the argument name when the `--ros-args` flag is added to the roslaunch command. Since information like `type`, `allowEmpty`, `help`, etc. are useful things to know about the arguments, it is nice to see them printed out with `--ros-args`. If we were to add this information as custom attributes, they would not be printed with `--ros-args`.

I found adding JSON objects inside the `doc` attributes to be the best way to add the information we need. I am happy to consider any alternatives you suggest.

Reply:
That seems fine, I didn't know ROS would complain about unknown arguments